test/parallel/test-timers-disarmed-in-callbak-not-fire-anymore.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var exec = require('child_process').exec;
+
+var testPath = './test/fixtures/test-timer-disarm.js';
+var testCommand = 'NODE_DEBUG=timer ' + process.execPath + ' ' + testPath;
+
+setTimeout(function() {
+  exec(testCommand, function(err, stdout, stderr) {
+    var o = stderr.split('\n');
+    var timer1Fired = 0;
+    var timer2Fired = 0;
+    var timer3Fired = 0;
+    for (var i = 0; i < o.length; i++) {
+      var line = o[i];
+      if (line.indexOf('timeout callback 11') >= 0) {
+        timer1Fired++;
+      }
+      if (line.indexOf('timeout callback 12') >= 0) {
+        timer2Fired++;
+      }
+      if (line.indexOf('timeout callback 13') >= 0) {
+        timer3Fired++;
+      }
+    }
+    assert.strictEqual(timer1Fired, 1);
+    assert.strictEqual(timer2Fired, 1);
+    assert.strictEqual(timer3Fired, 1);
+  });
+}, 100);
```

Comment:
Hmmm, fair enough. It seems slightly brittle, but I also don't know of a better way of it.